src/components/Map/NutritionChoropleth.tsx

```diff
@@ -0,0 +1,69 @@
+import { FeatureCollection, GeoJsonProperties, Geometry } from 'geojson';
+import L from 'leaflet';
+import React, { useEffect, useRef, useState } from 'react';
+import { GeoJSON, useMap } from 'react-leaflet';
+
+import { CountryMimiData } from '@/domain/entities/country/CountryMimiData';
+import NutritionChoroplethProps from '@/domain/props/NutritionChoroplethProps';
+import NutritionChoroplethOperations from '@/operations/map/NutritionChoroplethOperations';
+
+import NutritionStateChoropleth from './NutritionStateChoropleth';
+
+export default function NutritionChoropleth({
+  data,
+  countryId,
+  selectedCountryId,
+  selectedAlert,
+  setSelectedCountryId,
+  toggleAlert,
+  nutritionData,
+}: NutritionChoroplethProps) {
+  const geoJsonRef = useRef<L.GeoJSON | null>(null);
+  const map = useMap();
+
+  const [regionData, setRegionData] = useState<FeatureCollection<Geometry, GeoJsonProperties> | undefined>();
+  const [regionNutritionData, setRegionNutritionData] = useState<CountryMimiData | undefined>();
+  const [countryStyles, setCountryStyles] = useState<{ [key: number]: L.PathOptions }>({});
+
+  // given the `CountryNutrition` data -> parse the data to map country styling
+  useEffect(() => {
+    const parsedStyles = NutritionChoroplethOperations.getCountryStyles(nutritionData);
+    setCountryStyles(parsedStyles);
+  }, [nutritionData]);
+
+  if (selectedAlert) {
+    toggleAlert(selectedAlert);
```

Comment:
this should only happen on country click. Right now this disables alerts for the global view and what we want is when you click on a country and zoom in then alerts are disabled